lib/fetch/dataURL.js

```diff
@@ -604,14 +604,18 @@ function isHTTPWhiteSpace (char) {
  * @param {boolean} [trailing=true]
  */
 function removeHTTPWhitespace (str, leading = true, trailing = true) {
-  let i = 0; let j = str.length
+  let lead = 0
+  let trail = str.length - 1
+
   if (leading) {
-    while (j > i && isHTTPWhiteSpace(str.charCodeAt(i))) --i
+    while (lead < str.length && isHTTPWhiteSpace(str.charCodeAt(lead))) lead++
```

Comment:
It may be better to save the length.

Reply:
this code is very rarely run twice, it'd be a very pointless optimization (as in, I don't think I've ever seen a library or whatever put whitespaces in data urls).